torch_xla/csrc/xla_backend_impl.cpp

```diff
@@ -106,15 +107,16 @@ class XlaBackendImpl : public torch::lazy::BackendImplInterface {
     std::vector<torch::lazy::ComputationPtr> res;
     std::vector<xla::ComputationClient::CompileInstance> compile_instances;
     torch::lazy::BackendDevice current_device = GetCurrentDevice();
+    std::vector<xla::Shape> output_shapes;
 
     for (const torch::lazy::ComputationPtr instance : instances) {
       // TODO(JackCaoG): device is missing in instance, use CurrentDevice for
       // now
       const Computation* torch_xla_computation =
           dynamic_cast<Computation*>(instance.get());
-      xla::Shape shape = MakeShapeWithDeviceLayout(
+      output_shapes.push_back(MakeShapeWithDeviceLayout(
```

Comment:
Oh this is a c++ thing. If you look down below, we pass `&shape` to `xla::ComputationClient::Get()->Compile` through `compile_instances`. However if we only do 
```
      xla::Shape shape = MakeShapeWithDeviceLayout(
```

`shape` would only be on stack and get out of scope when this for loop ends. When `xla::ComputationClient::Get()->Compile` wants to access the shape it would be an invalid address. Therefore we need to keep shape in a vector and keep it alive until this function returns.